docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-set.md

```diff
@@ -5,39 +5,101 @@ sidebar_position: 4
 
 The [LightDB Set sample
 application](https://github.com/golioth/golioth-zephyr-sdk/tree/main/samples/lightdb/set)
-demonstrates how to send resources to LightDB State using the Golioth SDK. The
-process in non-blocking and after submitting a set request, the Golioth system
-client will complete the operation asyncronously.
+demonstrates how to send resources to LightDB State using the Golioth SDK. Data
+can be set synchronously, or asyncronously. Synchronous requests block program
+flow until a response is received. Asynchronous requests register a callback
+function that runs when a response is received at some point in the future.
 
-## Details
+## Includes
 
-### Payload preparation
+```c
+#include <net/golioth/system_client.h>
+```
+
+Including the Golioth System Client header file makes the LightDB Set API
+functions available to your program.
+
+## Payload preparation
 
 ```c
-char sbuf[64]; // Large enough for data plus zero terminator
+char sbuf[11]; // Large enough for data plus zero terminator
```

Comment:
Nit: prefer using the term "NULL terminator". They mean the same thing, but I think NULL terminator is more common.